### PR TITLE
created a QueueLogger for ServiceBus, this is very useful for bots th…

### PR DESCRIPTION
**Pull request by mzilberman**
…at need to post process messages.  Refactored Logger test to be useful in more than one logger.

this is a pre-review, **DO NOT MERGE**, just provide comments.

### CODE REVIEW

File: CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class AzureStorageQueueActivityLogger : IActivityLogger
+    {
+
+        private readonly JsonSerializerSettings _jsonSerializerSettings;
+        private CloudQueueClient _cloudQueueClient;
+        private CloudQueue _queueClient;
```
Comment (carlosscastro) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs]: I'd name this _queue or _cloudQueue to be consistent with line 18, and avoid confusions with the _cloudQueueClient which is the real client/

File: CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+using Microsoft.ServiceBus.Messaging;
+using Microsoft.WindowsAzure.Storage.Queue;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Azure
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class AzureStorageQueueActivityLogger : IActivityLogger
```
Comment (carlosscastro) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs]: Consider: AzureQueueActivityLogger would be a valid and shorter name. Not mandatory at all, just a thought

File: CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    {
+
+        private readonly JsonSerializerSettings _jsonSerializerSettings;
+        private CloudQueueClient _cloudQueueClient;
+        private CloudQueue _queueClient;
+        private QueueLoggerSettings _queueLoggerSettings;
+
+        private int _textMaxLength = 1024 * 20; //60k
+        private int _preCompressedMaxTextLength = 1024 * 50; //180k - assuming 3x compression
```
Comment (carlosscastro) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs]: Did you actually verify this? text compression is pretty powerful :)

File: CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="client"></param>
+        /// <param name="settings"></param>
+        public AzureStorageQueueActivityLogger(CloudQueue queueClient, QueueLoggerSettings loggerSettings = null, JsonSerializerSettings settings = null)
```
Comment (carlosscastro) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs]: Validate for null and throw ArgumentNullException if nul;l?

File: CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            //set the defaults
+            if (_queueLoggerSettings == null)
+                _queueLoggerSettings = new QueueLoggerSettings();
+            _queueClient = queueClient;
+            _jsonSerializerSettings = settings;
+        }
+
+        public async Task LogAsync(IActivity activity)
+        {
+
```
Comment (carlosscastro) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs]: remove extra space

File: CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueReader.cs [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    public class AzureStorageQueueReader: IQueueReader
+    {
+        private readonly CloudQueue _queueClient;
+        private QueueLoggerSettings _queueLoggerSettings;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="client"></param>
+        public AzureStorageQueueReader(CloudQueue client, QueueLoggerSettings queueSettings)
```
Comment (carlosscastro) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueReader.cs]: Validate null?

File: CSharp/Library/Microsoft.Bot.Builder.Azure/DomainTerminology.cs [hunk 7]
```diff
@@ -0,0 +1,86 @@
+﻿//using System;
+//using System.Collections.Generic;
+//using System.Linq;
+//using System.Text;
+//using System.Threading.Tasks;
+//using Autofac;
+//using Microsoft.Bot.Builder.Luis;
+
+//namespace Microsoft.Bot.Builder.Azure
```
Comment (carlosscastro) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/DomainTerminology.cs]: Pls no commented code. Can we remove this file?

File: CSharp/Library/Microsoft.Bot.Builder.Azure/QueueActivityModule.cs [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+        private string _serviceBusConnectionString = null;
+        private QueueLoggerSettings _queueLoggerSettings;
+
+        /// <summary>
+        /// Create a QueueLogger for a particular storage account and table name.
+        /// </summary>
+        /// <param name="connectionString">Azure ServiceBus connection string.</param>
+        /// <param name="queueName">Where to log activities.</param>
+        /// <param name="settings"></param>
+        public QueueActivityModule(CloudStorageAccount account, string queueName, QueueLoggerSettings loggerSettings= null, JsonSerializerSettings settings = null)
```
Comment (carlosscastro) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/QueueActivityModule.cs]: If users can call this, please thoroughly validate the parameters :)

File: CSharp/Library/Microsoft.Bot.Builder.Azure/ServiceBusActivityLogger.cs [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+    public enum LargeMessageMode
+    {
+        Discard,
+        Trim,
+        Error
+    }
+    /// <summary>
+    /// 
+    /// </summary>
+    public class ServiceBusActivityLogger : IActivityLogger
```
Comment (carlosscastro) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/ServiceBusActivityLogger.cs]: Cool to see this :)

File: CSharp/Library/Microsoft.Bot.Builder.Azure/ServiceBusActivityLogger.cs [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+        private int _textMaxLength = 1024*60; //60k
+        private int _preCompressedMaxTextLength = 1024*180; //180k - assuming 3x compression
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="client"></param>
+        /// <param name="settings"></param>
+        public ServiceBusActivityLogger(QueueClient client, QueueLoggerSettings loggerSettings = null, JsonSerializerSettings settings = null)
```
Comment (carlosscastro) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/ServiceBusActivityLogger.cs]: validate for null

File: CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs (hunk 2 above)
Comment (mzilberman) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs]: fixed

File: CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs (hunk 1 above)
Comment (mzilberman) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs]: fixed in both the Logger and Reader

File: CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs (hunk 3 above)
Comment (mzilberman) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs]: I am getting very weird compression perf.  on 100K message, compression was only down to 81K.  Lets discuss, numbers above are def not final.

File: CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs (hunk 4 above)
Comment (mzilberman) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs]: fixed in all places

File: CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs (hunk 5 above)
Comment (mzilberman) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueActivityLogger.cs]: fixed

File: CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueReader.cs (hunk 6 above)
Comment (mzilberman) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/AzureStorageQueueReader.cs]: fixed

File: CSharp/Library/Microsoft.Bot.Builder.Azure/DomainTerminology.cs (hunk 7 above)
Comment (mzilberman) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/DomainTerminology.cs]: sorry, removed

File: CSharp/Library/Microsoft.Bot.Builder.Azure/QueueActivityModule.cs (hunk 8 above)
Comment (mzilberman) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/QueueActivityModule.cs]: fixed everywhere in the code.

File: CSharp/Library/Microsoft.Bot.Builder.Azure/ServiceBusActivityLogger.cs (hunk 10 above)
Comment (mzilberman) [inline CSharp/Library/Microsoft.Bot.Builder.Azure/ServiceBusActivityLogger.cs]: fixed

**Comment by mzilberman** [comment]
All issues fixed, I will now start writing XML Comments

**Pull request closed without merging**